### ISSUE TITLE
Request for in21k-to-in1k finetuning script (hyperparameter)

### ISSUE THREAD

**Issue by hankyul2** [sail-sg/metaformer#11]
Hi @mavenlin @Trinkle23897 @yuweihao @ikuinen 

Thank you for sharing your awesome work!!

**Question:** I am wondering if you could share the in21k-to-in1k finetuning script (or details of hyperparameters). 

**Motivation:** I have tried to reproduce the `84.1%` top-1 accuracy of caformer-s18 pretrained in in21k and finetuned in in1k at 224 resolution, but could only achieve `83.4%` top-1 accuracy. Specifically, I downloaded the in21k-pretrained weight provided in this repo and finetuned it to the in1k dataset following the hyperparameters shown in the below table, which results in `83.4%` top-1 accuracy as shown in the below top-1 accuracy vs epoch graph. I will really appreciate it if you kindly provide us the details of the hyperparameter for the in21k-to-in1k finetuning script.

Table 9 of the [MetaFormer Baselines for Vision](https://arxiv.org/pdf/2210.13452.pdf): 
![image](https://github.com/sail-sg/metaformer/assets/31476895/cd78ce32-9624-4149-9bdb-6c3d240f4de5)

Experimental study of top-1 accuracy vs epoch for caformer-s18 finetuned at 224 resolution: 
![image](https://github.com/sail-sg/metaformer/assets/31476895/f9f4551f-725d-430b-b0ef-de3c82e9c781)

Thank you.

Hankyul.

**Comment by yuweihao** [comment]
Hi @hankyul2 , thanks for your attention.

1) I just added a script to process 21k head before 1k fine-tuning (see https://github.com/sail-sg/metaformer/blob/main/misc/process_21k_ckpt_before_1k_finetuning.py). This trick may let the model converge faster. 
2) There is a typo in the table. The learning for finetuning CAFormer should be `1e-4`. The finetuning script is like:
```
DATA_PATH=/path/to/imagenet
CODE_PATH=/path/to/code/metaformer # modify code path here
INIT_CKPT=/path/to/trained/model

ALL_BATCH_SIZE=1024
NUM_GPU=8
GRAD_ACCUM_STEPS=4 # Adjust according to your GPU numbers and memory size.
let BATCH_SIZE=ALL_BATCH_SIZE/NUM_GPU/GRAD_ACCUM_STEPS


cd $CODE_PATH && sh distributed_train.sh $NUM_GPU $DATA_PATH \
--model caformer_s18 --img-size 224 --epochs 30 --opt lamb --lr 1e-4 --sched None \
-b $BATCH_SIZE --grad-accum-steps $GRAD_ACCUM_STEPS \
--initial-checkpoint $INIT_CKPT \
--mixup 0 --cutmix 0 \
--model-ema --model-ema-decay 0.9999 \
--drop-path 0.1 --head-dropout 0.2
```

**Comment by hankyul2** [comment]
Thank you for your quick reply!

I will finetune the model following your suggestion and share the result here as soon as the train done.

Hankyul.
